### PR TITLE
Make the seed node port dynamic and functionality for fetching micro block

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: src/libServer/Server.cpp [hunk 1]
```diff
@@ -1106,3 +1106,49 @@ string Server::GetNumTxnsDSEpoch() {
...
+  if (txBlock == TxBlock()) {
+    throw JsonRpcException(RPC_INVALID_PARAMS, "Tx Block does not exist");
+  }
+
+  auto microBlockInfos = txBlock.GetMicroBlockInfos();
+
+  for (auto const& mbInfo : microBlockInfos) {
+    MicroBlockSharedPtr mbptr;
+    if (mbInfo.m_txnRootHash == TxnHash() && mbInfo.m_shardId == shardID) {
+      throw JsonRpcException(RPC_INVALID_PARAMETER, "Shard has no txns empty");
```
Comment (ansnunez) [inline src/libServer/Server.cpp]: Seems like `RPC_MISC_ERROR` is a better code? Also, maybe can just say "Shard has no txns"

File: src/libPersistence/BlockStorage.cpp [hunk 2]
```diff
@@ -133,6 +133,20 @@ bool BlockStorage::GetTxnFromHistoricalDB(const dev::h256& key,
   return true;
 }
 
+bool BlockStorage::GetHistoricalMicroBlock(const BlockHash& blockhash,
```
Comment (ansnunez) [inline src/libPersistence/BlockStorage.cpp]: Is there supposed to be a `PutHistoricalMicroBlock`, or any code that stores into `m_MBHistoricalDB`?

File: src/libZilliqa/Zilliqa.cpp [hunk 3]
```diff
@@ -281,6 +281,8 @@ Zilliqa::Zilliqa(const PairOfKey& key, const Peer& peer, bool loadConfig,
       case SyncType::DB_VERIF:
         LOG_GENERAL(INFO, "Intitialize DB verification");
         m_n.ValidateDB();
+        std::this_thread::sleep_for(std::chrono::seconds(10));
+        raise(SIGKILL);
```
Comment (ansnunez) [inline src/libZilliqa/Zilliqa.cpp]: Doesn't `ValidateDB` already call `raise(SIGKILL)`?

Comment (KaustubhShamshery) [reply src/libZilliqa/Zilliqa.cpp]: no I removed it from there as the function might return false and then it is not invoked

File: src/libServer/Server.cpp (hunk 1 above)
Comment (KaustubhShamshery) [reply src/libServer/Server.cpp]: yeah i think i will replace it with database error

File: src/libPersistence/BlockStorage.cpp (hunk 2 above)
Comment (KaustubhShamshery) [reply src/libPersistence/BlockStorage.cpp]: actually that is done by backend using s3

**Pull request merged**